include/ginkgo/core/factorization/cholesky.hpp

```diff
@@ -67,6 +67,17 @@ class Cholesky
          * incorrect results or crash.
          */
         bool GKO_FACTORY_PARAMETER_SCALAR(skip_sorting, false);
+
+        /**
+         * If the user provides the symbolic factorization, it should contain
+         * the fill-in for the matrix. i.e., When this is true, the symbolic
+         * factorization must contain the non-zero locations in the original
+         * matrix and the corresponding fill-in locations during factorization.
+         * If it does not have full fill-in, as in Ilu, this parameter must be
+         * set to false in order to avoid the possibility of hanging or illegal
+         * memory accesses during the factorization process.
+         */
+        bool GKO_FACTORY_PARAMETER_SCALAR(full_fillin, true);
```

Comment:
Why this parameter? This goes against the name LU, and could as well be handled by ILU, if we hand in the sparsity pattern there as a new parameter.

Reply:
does it go against LU? I try to avoid copying half of functions from lu to ilu.
We do not need the sparsity pattern in ilu currently because we use the same sparsity of the matrix

Reply:
In its current form they use the same algorithm, but adding this parameter prevents future evolution taking advantage of structural properties of an LU that an ILU doesn't have.

Reply:
could you elaborate more?
@nbeams I assume you use the ILU on your side now not the LU interface, right?
I will copy some function into ILU.
It should still work with ILU-only interface for the case we discussed above in the future

Reply:
have copied